workers/data_refinery_workers/downloaders/sra.py

```diff
@@ -297,11 +311,21 @@ def download_sra(job_id: int) -> None:
 
     original_file = original_files.first()
     sample = original_file.samples.first()
-    if _has_unmated_reads(sample.accession_code):
-        original_files = _replace_dotsra_with_fastq_files(sample, job, original_file)
 
     downloaded_files = []
     success = None
+
+    # Try accessing FTP server, see if that was successful
+    unmated_reads, success = _has_unmated_reads(sample.accession_code, job)
```

Comment:
Hmm, I prefer to use functions with multiple return values as little as possible. I think that maybe you could just remove the first try-catch block from the above function and wrap this function call with it instead. That way this function handles failure to login, and then if the function doesn't raise an exception then the return value can be used to indicate the presence/absence of the files.